Core/Util/PortManager.cpp

```diff
@@ -54,16 +54,14 @@ PortManager::PortManager():
 	m_InitState(UPNP_INITSTATE_NONE),
 	m_LocalPort(UPNP_LOCAL_PORT_ANY),
 	m_leaseDuration("43200") {
-	// Since WSAStartup can be used multiple times it should be safe to do this right?
-	net::Init();
+	// Don't call net::Init or similar here, we don't want stuff like that to happen before main.
 }
 
 PortManager::~PortManager() {
 	// FIXME: On Windows it seems using any UPnP functions in this destructor that gets triggered when exiting PPSSPP will resulting to UPNPCOMMAND_HTTP_ERROR due to early WSACleanup (miniupnpc was getting WSANOTINITIALISED internally)
 	Clear();
 	Restore();
 	Terminate();
-	net::Shutdown();
```

Comment:
As i remembered not using `net::Shutdown` at the last line here (paired with net::Init at constructor above) will cause UPnPManager failed to clean up opened ports if the player closing PPSSPP with the [X] button while playing a game. May be also causing a timeout being reached that will slowdown the exit process, due to WSACleanup being called before reaching this destructor, may need to ensure that PortManager being destroyed before that WSACleanup to avoid issue.

Reply:
Right, I think I'll have to figure out a better way here. Think I'll call PortManager init/deinit from NativeInit and NativeShutdown instead of using constructor/destructor.